app/controllers/katello/api/v2/content_view_versions_controller.rb

```diff
@@ -82,9 +82,14 @@ def update
 
     api :PUT, "/content_view_versions/:id/republish_repositories", N_("Forces a republish of the version's repositories' metadata")
     param :id, :number, :desc => N_("Content view version identifier"), :required => true
-    param :force, :bool, :desc => N_("Force metadata regeneration to proceed. (Deprecated)"), deprecated: true
+    param :force, :bool, :desc => N_("Force metadata regeneration to proceed. Dangerous operation when version has repositories with the 'Complete Mirroring' mirroring policy")
     def republish_repositories
-      task = async_task(::Actions::Katello::ContentViewVersion::RepublishRepositories, @content_view_version)
+      mirror_complete_repos = @content_view_version.repositories.joins(:root).where(root: { mirroring_policy: ::Katello::RootRepository::MIRRORING_POLICY_COMPLETE })
```

Comment:
It might need to be a different kind of join, to exclude repos outside the CV?

Reply:
Hmm..Not sure I can reproduce. Did you not have those repos in the CV version?

Reply:
Nope, only the custom repos were added.

Reply:
I was using the wrong id.. ignore me :see_no_evil: